packages/issue-receiver/src/main.ts

```diff
@@ -0,0 +1,48 @@
+import * as core from '@actions/core'
+import * as gh from '@actions/github'
+import * as Webhooks from '@octokit/webhooks'
+
+// TODO: Generalise this to work on any object
+interface Notification {
+    'list-ops': Exercise
+}
+
+interface Exercise {
+    commit_sha: string
+    commit_message: string
+    pull_request_html_url: string
+    new_cases: string[]
+}
+
+async function run(): Promise<void> {
+    try {
+        // Confirm that it's a repository_dispatch event
+        if (gh.context.eventName != 'repository_dispatch') {
+            throw new Error(`Event ${gh.context.eventName} is not supported`)
+        }
+
+        // Init octokit
+        const octokit = gh.getOctokit(core.getInput('token'))
+
+        const eventPayload = gh.context.payload as Webhooks.EventPayloads.WebhookPayloadRepositoryDispatch
```

Comment:
Just for my information, as I've not checked this out on my computer(s), is gh.context.payload untyped?

Or is it a union type and you're trying to narrow it here to a single thing?

Reply:
AFAIU it's a union. There's one such type for each webhook event (of which there are dozens).

I took this from the `@actions/github` docs: https://github.com/actions/toolkit/tree/main/packages/github#webhook-payload-typescript-definitions

Reply:
Sigh, got it!. TS is smart enough to do this itself. I'll fix it when you're done with other things.